python/ray/tune/ray_trial_executor.py

```diff
@@ -206,34 +204,47 @@ def start_trial(self, trial, checkpoint=None):
                 of trial.
         """
         self._commit_resources(trial.resources)
-        try:
-            self._start_trial(trial, checkpoint)
-        except AbortTrialExecution:
-            logger.exception("Trial %s: Error starting runner, aborting!",
-                             trial)
-            time.sleep(2)
-            error_msg = traceback.format_exc()
-            self._stop_trial(trial, error=True, error_msg=error_msg)
-            return  # don't retry fatal Tune errors
-        except Exception:
-            logger.exception(
-                "Trial %s: Error starting runner. Attempting "
-                "restart without checkpoint.", trial)
-            time.sleep(2)
-            error_msg = traceback.format_exc()
-            self._stop_trial(trial, error=True, error_msg=error_msg)
+        remote_runner = None
+        attempts = 0
+        while attempts < TRIAL_START_ATTEMPTS:
+            attempts += 1
+            if attempts > 1:
+                logger.warning("Trial %s: Start attempt #%s...", trial,
+                               attempts)
             try:
-                # This forces the trial to not start from checkpoint.
-                trial.clear_checkpoint()
-                self._start_trial(trial)
+                self._start_trial(trial, checkpoint, remote_runner)
+                break
+            except AbortTrialExecution:
+                logger.exception("Trial %s: Error starting runner, aborting!",
+                                 trial)
+                time.sleep(2)
+                error_msg = traceback.format_exc()
+                self._stop_trial(trial, error=True, error_msg=error_msg)
+                break  # don't retry fatal Tune errors
+            except RayTimeoutError:
+                logger.warning(
+                    "Trial %s: Runner task timed out. This could be due to "
+                    "slow worker startup.", trial)
```

Comment:
Can this error message be more informative? Anything actionable? Otherwise, is Tune taking further actions that I need to know about?

Reply:
ie., it is reusing the remote_runner?